src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -85,6 +85,7 @@ class PhpDumper extends Dumper
     private array $singleUsePrivateIds = [];
     private array $preload = [];
     private bool $addGetService = false;
+    private bool $addContainerRef = false;
```

Comment:
Looking at changes in test fixtures, I see a bunch of cases where the new generated code adds `$this->ref` without ever using it. Shouldn't it also be controlled by that flag ?

Reply:
That code is cheap, I thought it wouldn't provide much :shrug: